### DISCUSSION TITLE
Access Http context in the class

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#3280] [Help] [answered]
I have a simple class like below and need to acces `http context` in the `verify` method

```javascript
import HttpContext from '@ioc:Adonis/Core/HttpContext'

class Test {
  public static verify() {
     const ctx = HttpContext.get()

      console.log(ctx) // null
  }
}
``` 

I need to access the `request` in this method.

**Comment by MaximeMRF** [comment, marked as answer]
Hello,
I think you should read the [Controllers guide](https://docs.adonisjs.com/guides/controllers).
If you want to access the `http context` the following code will help you.

```js
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class Test {
  public static verify(ctx: HttpContextContract) {
    console.log(ctx)
  }
}
```
Or if you want just to access the `request`:

```js
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class Test {
  public static verify({ request }: HttpContextContract) {
    console.log(request)
  }
}
```

**Reply by MaximeMRF** [reply]
I think this solution is better for you:

If you want to use `HttpContext.get()` enable the async local storage in `config/app.ts`

Read [this](https://docs.adonisjs.com/guides/async-local-storage#usage) part of the guide for more informations.
